proto/storage/notifier.proto

```diff
@@ -151,9 +151,7 @@ message MicrosoftSentinel {
     bool enabled = 3;
   }
   DataCollectionRuleConfig alert_dcr_config = 5;
```

Comment:
Does this need a one_of or the same instance of the notifier can have different rules for alert and audit? Assume the latter?

Reply:
Yeah I think I agree.  `one_of` seems to make sense here and I don't even like the `one_of` construct.

Reply:
Unless ALL these can be applicable at the same time.  And I'm not sure what that would look like.

Reply:
> Does this need a one_of or the same instance of the notifier can have different rules for alert and audit? Assume the latter?

Yes, both can be valid at the same time. 

**Workflow**

A customer configures the Microsoft Sentinel integration. The customer needs to send the data and process it to make it available for analytics in Microsoft Sentinel, this is done by the data_collection_rule.

The data_collection_rule can be different for each data source which contains logic to parse, for example it looks like this:

```
source
| extend policy_name=parse_json(msg).policy.name
| extend cluster_name=parse_json(msg).clusterName
| extend namespace_id=parse_json(msg).deployment.namespaceId
```

![Screenshot from 2024-09-10 11-27-38](https://github.com/user-attachments/assets/585bbaf3-1de9-4b1a-b6c9-9f902de42386)